cpu/sam0_common/periph/i2c.c

```diff
@@ -230,180 +187,79 @@ int i2c_release(i2c_t dev)
     return 0;
 }
 
-int i2c_read_byte(i2c_t dev, uint8_t address, void *data)
-{
-    return i2c_read_bytes(dev, address, data, 1);
-}
-
-int i2c_read_bytes(i2c_t dev, uint8_t address, void *data, int length)
+int i2c_read_bytes(i2c_t dev, uint16_t addr,
+                   void *data, size_t len, uint8_t flags)
 {
-    SercomI2cm *i2c;
-
-    switch (dev) {
-#if I2C_0_EN
-        case I2C_0:
-            i2c = &I2C_0_DEV;
-            break;
-#endif
-        default:
-            return -1;
+    if (dev >= I2C_NUMOF) {
+        return -1;
     }
-
-    /* start transmission and send slave address */
-    if (_start(i2c, address, I2C_FLAG_READ) < 0) {
-        return 0;
+    if (!(flags & I2C_NOSTART))
```

Comment:
Missing opening `{`, the if block should be surrounded with `{}` even if it's a one liner.  It improves readability.

Reply:
Oops ...
Nice catch !